### DISCUSSION TITLE
Add ON DELETE CASCADE on child tables

### DISCUSSION THREAD

**Discussion by tbobker123** [codeigniter4/shield#511] [Ideas]
to remove a user, you need to delete all rows related to the user from other tables as the example shows below.  Would it be a better idea to add the ON DELETE CASCADE feature to all child tables so you only have to delete the user from the users table?

```
DELETE FROM users WHERE id = 1;
DELETE FROM auth_groups_users WHERE user_id = 1;
DELETE FROM auth_identities WHERE user_id = 1;
DELETE FROM auth_logins WHERE user_id = 1;
DELETE FROM auth_permissions_users WHERE user_id = 1;
DELETE FROM auth_remember_tokens WHERE user_id = 1;
DELETE FROM auth_token_logins WHERE user_id = 1;
```

**Comment by MGatner** [comment]
I prefer to let my code manage all my persistence logic, otherwise you get tightly coupled to _how_ your data is persisted. Things like foreign keys and not-nullable fields make it harder to generate interesting test data or do imports and other migrations. This leads to the bad habit of trusting persistence data in your code: "this column is INT NOT NULL so I don't need to check `is_numeric()` or cast it". In my experience these kind of things always lead to strange bugs down the road. 

I know I'm not in the majority so I will let the team decide on this one: just my two cents.

**Comment by kenjis** [comment]
We have already set `ON DELETE CASCADE` for foreign keys.
See https://github.com/codeigniter4/shield/blob/7e90fdaee854c14a7b53b0384aa27de6da5d4cfd/src/Database/Migrations/2020-12-28-223112_create_auth_tables.php#L50
